### ISSUE TITLE
Batch selections should not preview single-element non-splats

### ISSUE THREAD

**Issue by jleibs** [rerun-io/rerun#1014]
Although it's not easy to do with our APIs at the moment, it's theoretically possible to log a non-splatted element with a single instance-key.

For example, a future API that allows us to log colors directly:
```
rr.log_points("points", [[1,2],[3,4]], instance_keys=[1, 2])
rr.log_colors("points", [RED], instance_keys=[2])
```

In this case, color is not a splat, but if you hover on the *batch* of points in the UI, it will display as if RED applies to the whole batch rather than only applying to the 2nd point.

**Comment by teh-cmc** [comment]
This should be irrelevant now that splats are handled in independent rows? @jleibs

**Comment by jleibs** [comment]
> This should be irrelevant now that splats are handled in independent rows? @jleibs

Nope, still an issue. This snippet is actually runnable:
```
import rerun as rr

rr.init("point_test", spawn=True)
rr.log_points("points", positions=[[1, 2], [3, 4]], identifiers=[1, 2])
rr.log_points("points", positions=None, colors=[[255, 0, 0]], identifiers=[2])
```

Produces this data UI:
![image](https://user-images.githubusercontent.com/3312232/221921051-3b667cb3-ecce-40af-bde5-f7d8c146eba4.png)

The selection UI makes it *look* like all the points should be red, but since the 2nd log-value is not a splat, only the second point is actualy red (which is correct)

I think the issue is basically here, where we should explicitly check for a splat instead of a length of 1.
https://github.com/rerun-io/rerun/blob/jleibs/tensor_view_without_classic/crates/re_viewer/src/ui/data_ui/component.rs#L52

**Comment by teh-cmc** [comment]
Should be fixed by #1619
